### PR TITLE
Created AppLoader component

### LINKED ISSUE #62: Create AppLoader component
Create loader component with color variants, size variants

### PR DESCRIPTION
**Pull request by D1nviz**
Created a custom loader component with variants.

https://github.com/idx-academy/spa-orders/assets/118522791/8249a813-76ff-4551-88b6-5b4ea054ef0b

important here is required to override MUI inline styles
image

### CODE REVIEW

File: code/src/components/app-loader/AppLoader.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  @each $variant, $colors in $loader-colors {
+    &__#{$variant} {
+      @include loader-color(nth($colors, 1));
+    }
+  }
+
+  @each $size, $values in $loader-sizes {
+    &__#{$size} {
+      width: nth($values, 1) !important;
+      height: nth($values, 1) !important;
```
Comment (mhevyk) [inline code/src/components/app-loader/AppLoader.scss:50]: Add comment for future developers to inform about thr reason of usage `!important`

File: code/src/components/app-loader/AppLoader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    expect(loader).toBeInTheDocument();
+  });
+
+  test("renders AppLoader with additional class names", () => {
+    render(<AppLoader className="custom-class" />);
+    const loader = screen.getByRole("progressbar");
+    expect(loader).toBeInTheDocument();
+    expect(loader).toHaveClass("spa-loader");
+    expect(loader).toHaveClass("custom-class");
+  });
```
Comment (mhevyk) [inline code/src/components/app-loader/AppLoader.test.tsx]: Is is nessesary test? One test is not enough?

File: code/src/components/app-loader/AppLoader.scss (hunk 1 above)
Comment (D1nviz) [reply code/src/components/app-loader/AppLoader.scss:50]: Added comment

File: code/src/components/app-loader/AppLoader.test.tsx (hunk 2 above)
Comment (D1nviz) [reply code/src/components/app-loader/AppLoader.test.tsx]: removed

**Review by mhevyk — APPROVED** [review]

**Review by ValKoDim — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Pull request merged**